crates/stc_ts_type_checker/tests/conformance/expressions/functionCalls/typeArgumentInference.stats.rust-debug

```diff
@@ -1,6 +1,6 @@
 Stats {
     required_error: 2,
     matched_error: 2,
-    extra_error: 1,
+    extra_error: 2,
```

Comment:
As far as I can tell this is fine, because call-site type param inference isn't implemented yet.

```ts
function someGenerics8<T>(n: T): T { return n; }
var x = someGenerics8(someGenerics7); // T can't be inferred to be `someGenerics7` here, 
          // thus the return type is T and not the function `someGenerics7` (afaik)
                                                                        
```

Reply:
It's implemented

Reply:
Related:

https://github.com/dudykr/stc/blob/508b2bab619e616350332fc1f845879d5506c6c7/crates/stc_ts_file_analyzer/src/analyzer/expr/call_new.rs#L2637-L2651

Reply:
That works thanks, there's another issue, it seems like in some cases the type params are "lost"

```ts
function x<T extends string>(a: T) {
  return <T, U>() => {};
}
var y = x("hello"); // f.type_params -> correct
y<string, number>(); // f.type_params -> None
```

so thats what Im going to try and figure out, not sure if ill be able to or how long.

Reply:
I think it's fine to fix it with another PR

Reply:
Feel free to review then... You can also consider blocking this PR because once the inference for this case is fixed it will probably require changes in `get_return_type`